cli/run_util.go

```diff
@@ -467,6 +473,10 @@ func (r WorkflowRunner) runStep(
 		return 1, []envmanModels.EnvironmentItemModel{}, err
 	}
 
+	if err := exportEnvKeyList(secretKeys); err != nil {
```

Comment:
Thinking if it would be better to move this before runStep(), it seems not dependant on anything else here. That would reduce the already long function argument list.
The next possible followup after that (maybe in a seperate PR) would be to collect all the secret handling/preparation into a new function (reduce the lines in run_util :) ).

Reply:
That is a great tip and I have implemented it.